src/states/elder/mod.rs

```diff
@@ -714,24 +714,31 @@ impl Elder {
             msg.routing_message()
         );
 
-        let dst = msg.message_dst();
-        let signed_msg = msg.signed_routing_message();
-        if self.in_location(dst) {
-            self.check_signed_message_trust(signed_msg)?;
-            self.check_signed_message_integrity(signed_msg)?;
-            self.update_our_knowledge(signed_msg);
-
-            if dst.is_multiple() {
-                // Broadcast to the rest of the section.
-                if let Err(error) = self.send_signed_message(&msg) {
-                    debug!("{} Failed to send {:?}: {:?}", self, signed_msg, error);
-                }
+        let in_location = self.in_location(msg.message_dst());
+        if !in_location || msg.message_dst().is_multiple() {
+            // Relay closer to the destination or broadcast to the rest of our section.
+            if let Err(error) = self.send_signed_message(&msg) {
+                debug!(
+                    "{} Failed to send {:?}: {:?}",
+                    self,
+                    msg.signed_routing_message(),
+                    error
+                );
+            }
+        }
+
+        if in_location {
+            let signed_msg = msg.signed_routing_message();
+            let handle = self.verify_signed_message(signed_msg).map_err(|error| {
+                self.log_verify_failure(signed_msg, &error);
+                error
+            })?;
+
+            if handle {
+                self.update_our_knowledge(signed_msg);
+                self.routing_msg_queue
+                    .push_back(msg.into_signed_routing_message());
```

Comment:
what's the difference between `msg.signed_routing_message` (https://github.com/maidsafe/routing/pull/2018/files#diff-bb7fddc1a592903612286af78e19ad6bR731) and `msg.into_signed_routing_message` ?

Reply:
One returns reference the other value.

Reply:
(one borrows, the other moves)